src/umpire/Replay.hpp

```diff
@@ -18,29 +18,16 @@
 #include <string>
 #include <sstream>
 #include <iostream>
-#include "umpire/Allocator.hpp"
-
-#define UMPIRE_REPLAY( msg )                                             \
-{                                                                        \
-  if (umpire::replay::Replay::getReplayLogger()->replayLoggingEnabled()) { \
-    std::ostringstream local_msg;                                        \
-    local_msg  << "REPLAY," << msg;                                      \
-    umpire::replay::Replay::getReplayLogger()->logMessage(local_msg.str());\
-  }                                                                      \
-}
 
-#define UMPIRE_REPLAY_CONT( msg )                                        \
-{                                                                        \
-  if (umpire::replay::Replay::getReplayLogger()->replayLoggingEnabled()) { \
-    std::ostringstream local_msg;                                        \
-    local_msg  << "," << msg;                                            \
-    umpire::replay::Replay::getReplayLogger()->logMessage(local_msg.str());\
-  }                                                                      \
-}
+#include "umpire/Allocator.hpp"
+#include "umpire/strategy/AllocationStrategy.hpp"
+#include "umpire/strategy/DynamicPool.hpp"
 
 namespace umpire {
 namespace replay {
-std::ostream& operator<< (std::ostream& out, umpire::Allocator& alloc);
+std::ostream& operator<< (std::ostream& out, umpire::Allocator& );
+std::ostream& operator<< (std::ostream& out, 
```

Comment:
Do we still need this, given that you are trying to use `typeid` below to skip printing the heuristic?

Reply:
Yes, even though I am using the typeid, the code will still exist within the if block and the compiler will complain that the operator does not know how to handle that type.  So, the code is protected by the if block and will never be called, but the code itself still exists.

Reply:
Gotcha.